src/rc_string.c

```diff
@@ -108,11 +109,61 @@ IMPLEMENT_MOCKABLE_FUNCTION(, THANDLE(RC_STRING), rc_string_create, const char*,
 }
 
 
+
 THANDLE(RC_STRING) rc_string_create_with_format(const char* format, ...)
 {
-    (void)format;
+    THANDLE(RC_STRING) result = NULL;
 
-    return NULL;
+    if (format == NULL)
+    {
+        /*Codes_SRS_RC_STRING_07_001: [ If format is NULL, rc_string_create_with_format shall fail and return NULL. ]*/
+        LogError("Invalid arguments: const char* format=%s", MU_P_OR_NULL(format));
+    }
+    else
+    {
+        /*Codes_SRS_RC_STRING_07_002: [ Otherwise, `rc_string_create_with_format` shall determine the total number of characters written using the variable number of arguments. ]*/
+        va_list args;
+        va_start(args, format);
+        size_t string_length_with_terminator = vsnprintf(NULL, 0, format, args) + 1;
+        
+        if (string_length_with_terminator <= 0)
+        {
+            /*Codes_SRS_RC_STRING_07_003: [ If `vsnprintf` failed to determine the total number of characters written, `rc_string_create_with_format` shall fail and return `NULL`. ]*/
+            LogError("vsnprintf failed to determine the total number of characters written.");
+        }
+        else
+        {
+            /*Codes_SRS_RC_STRING_07_004: [ `rc_string_create_with_format` shall allocate memory for the `THANDLE(RC_STRING)`and the number of bytes for the resulting formatted string. ]*/
+            THANDLE(RC_STRING) temp_result = THANDLE_MALLOC_WITH_EXTRA_SIZE(RC_STRING)(rc_string_dispose, sizeof(RC_STRING_INTERNAL) - sizeof(RC_STRING) + string_length_with_terminator);
```

Comment:
>sizeof(RC_STRING_INTERNAL) - sizeof(RC_STRING) + string_length_with_terminator [](http://example.com/codeflow?start=106&length=78)

Funny thing is that "in theory" this can overflow.